pkg/controller/migration/convert/core.go

```diff
@@ -9,6 +13,50 @@ func handleCore(c *components, install *Installation) error {
 		return ErrIncompatibleCluster{"only DATASTORE_TYPE=kubernetes is supported at this time"}
 	}
 
+	// node resource limits
+	node := getContainer(c.node.Spec.Template.Spec, "calico-node")
+	if len(node.Resources.Limits) > 0 || len(node.Resources.Requests) > 0 {
+		install.Spec.ComponentResources = append(install.Spec.ComponentResources, &operatorv1.ComponentResource{
+			ComponentName:        operatorv1.ComponentNameNode,
+			ResourceRequirements: node.Resources.DeepCopy(),
+		})
+	}
+
+	// kube-controllers
+	kubeControllers := getContainer(c.kubeControllers.Spec.Template.Spec, "calico-kube-controllers")
+	if len(kubeControllers.Resources.Limits) > 0 || len(kubeControllers.Resources.Requests) > 0 {
+		install.Spec.ComponentResources = append(install.Spec.ComponentResources, &operatorv1.ComponentResource{
+			ComponentName:        operatorv1.ComponentNameKubeControllers,
+			ResourceRequirements: kubeControllers.Resources.DeepCopy(),
+		})
+	}
+
+	// typha resource limits. typha is optional so check for nil first
+	typha := getContainer(c.typha.Spec.Template.Spec, "calico-typha")
+	if typha != nil && (len(typha.Resources.Limits) > 0 || len(typha.Resources.Requests) > 0) {
+		install.Spec.ComponentResources = append(install.Spec.ComponentResources, &operatorv1.ComponentResource{
+			ComponentName:        operatorv1.ComponentNameTypha,
+			ResourceRequirements: typha.Resources.DeepCopy(),
+		})
+	}
+
+	// kube-controllers nodeSelector
+	install.Spec.ControlPlaneNodeSelector = c.kubeControllers.Spec.Template.Spec.NodeSelector
+
+	// node update-strategy
+	install.Spec.NodeUpdateStrategy = c.node.Spec.UpdateStrategy
+
+	// node volumes
+	for _, vol := range c.node.Spec.Template.Spec.Volumes {
+		switch vol.Name {
+		case "flexvol-driver-host":
+			if vol.HostPath == nil {
+				return ErrIncompatibleCluster{"volume 'flexvol-driver-host' must be a HostPath"}
+			}
+			install.Spec.FlexVolumePath = vol.HostPath.Path
+		}
+	}
+
```

Comment:
I think we should check if install.Spec.FlexVolumePath is not set at this point then we should set it to none. Otherwise it will get defaulted later.

Reply:
We might also want to verify that the flexvol init container doesn't exist in the case when we set FlexVolumePath to None.